src/components/search/tests/SearchCourseCard.test.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { AppContext } from '@edx/frontend-platform/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import SearchCourseCard from '../SearchCourseCard';
+
+import { renderWithRouter } from '../../../utils/tests';
+
+const TEST_ENTERPRISE_SLUG = 'test-enterprise-slug';
+
+jest.mock('react-truncate', () => ({
+  __esModule: true,
+  default: ({ children }) => children,
+}));
+
+jest.mock('react-loading-skeleton', () => ({
+  __esModule: true,
+  default: () => <span data-testid="did-i-render" />,
+}));
+
+const SearchCourseCardWithAppContext = (props) => (
+  <AppContext.Provider
+    value={{
+      enterpriseConfig: { slug: TEST_ENTERPRISE_SLUG },
+    }}
+  >
+    <SearchCourseCard {...props} />
+  </AppContext.Provider>
+);
+
+const TEST_COURSE_KEY = 'test-course-key';
+const TEST_TITLE = 'Test Title';
+const TEST_CARD_IMG_URL = 'http://fake.image';
+const TEST_PARTNER_NAME = 'Partner Name';
+
+const defaultProps = {
+  hit: {
+    key: TEST_COURSE_KEY,
+    title: TEST_TITLE,
+    card_image_url: TEST_CARD_IMG_URL,
+    partners: [TEST_PARTNER_NAME],
+  },
+};
+
+const propsForLoading = { ...defaultProps, isLoading: true };
+
+describe('<SearchCourseCard />', () => {
+  test('renders the correct data', () => {
+    const { container } = renderWithRouter(<SearchCourseCardWithAppContext {...defaultProps} />);
+
+    expect(screen.getByText(TEST_TITLE)).toBeInTheDocument();
+    expect(screen.getByAltText(TEST_PARTNER_NAME)).toBeInTheDocument();
+
+    expect(container.querySelector('.discovery-card > a')).toHaveAttribute(
+      'href',
+      `/${TEST_ENTERPRISE_SLUG}/course/${TEST_COURSE_KEY}`,
+    );
+    expect(container.querySelector('p.partner')).toHaveTextContent(TEST_PARTNER_NAME);
+    expect(container.querySelector('.card-img-top')).toHaveAttribute('src', TEST_CARD_IMG_URL);
+  });
+
+  test('renders the loading state', () => {
+    renderWithRouter(<SearchCourseCardWithAppContext {...propsForLoading} />);
+
+    // assert the mocked out loading component renders exactly 5 times to verify
+    // course card is properly in a loading state
+    expect(screen.queryAllByTestId('did-i-render')).toHaveLength(5);
```

Comment:
Is this making sure that 5 course card skeletons were rendered or just one card skeleton that was rendered 5 times? If it's 5 skeleton cards is that 5 amount tied to a constant somewhere? If it's one skeleton being rendered 5 times why is it being rendered exactly 5 times?

Reply:
This is for an individual course card, that when in a loading state, shows 5 `<Skeleton />` components, which is the number of times that component is currently rendered when a course card is in the loading state.

Each `<Skeleton />` is for the following:
1. card image
2. partner's logo
3. partner's name
4. course title
5. "Course" descriptor at the bottom

<img width="330" src="https://user-images.githubusercontent.com/2828721/86384222-6d995580-bc5c-11ea-985b-a38b1b577c68.png" />

Reply:
As for whether that 5 should be tied to a constant, the constant would only be used in this one place, and is specific to the number of times `<Skeleton />` is rendered. I guess I'm not sure really the value of a constant here.

Reply:
Either way, I will add it a constant 🤷‍♂️

Reply:
Well the constant part was for whether it was like an initial 5 cards displayed on a page on initial load type thing. My only concern right now is the design of the card changing and the number of skeletons changing because of it causing this test to fail and the person making a change not quite being able to understand why.

What I've seen done before is there being a completely separate Skeleton component that is displayed on the page/grid instead of the loaded Card component. Which reduces the number of `isLoading` if statements down to being just one because the skeleton part is at a higher level component. I'll DM you an example.

I'm not saying "hey let's change up how skeletons are done right now cuz people in the future could be confused by this test breaking" cuz that would be ridiculous, but I think maybe we could change up this test to not expect there to be exactly 5 skeletons and maybe expect there to be to be at least the most core ones like the image, the course title, and the partner's name.

Reply:
@kellmarie Ah yes, that is fair about not expecting exactly 5 skeletons but rather checking for the primary ones like you mentioned. I will update 👍 

There is a separate `SearchCourseCard.Skeleton` that gets rendered but all it does it add the `isLoading` prop to the base `SearchCourseCard` component. The reason I did it this way is to not duplicate the DOM-related things between a skeleton course card component and the normal skeleton course card component. Figured it'd be better to have the inline `isLoading` conditionals.